### ISSUE TITLE
Better interface for marking checks as paid.

### ISSUE THREAD

**Issue by harrislapiroff** [littleweaver/django-brambling#287]
The current button isn't obviously a button, but perhaps more importantly, there should be probably be a box where you can enter an order number and mark it as paid, rather than having to pull up that order's detail view from the table.

**Comment by melinath** [comment]
I don't think it would be very user-friendly to have a text field that you put the order's code in. Maybe as an admin action-like thing?

**Comment by harrislapiroff** [comment]
@melinath Doesn't an organizer already have to put the order code in to search for it?

**Comment by harrislapiroff** [comment]
Actually, wait, there's no way for an organizer to find the correct order right now?

**Comment by harrislapiroff** [comment]
I think a field for searching by order code is going to be a necessary step to find the correct order to apply a payment to for any event that has more than, say, 50 orders. I can't think of any more user-friendly way to locate the correct order, though I'm open to ideas.

Creating a form that accepted an order code and marked an order as paid would just reduce the task of accepting a check to a single step instead of searching for the correct order, viewing it's details, and marking it as paid.

**Comment by nmorduch** [comment]
What I would really like to see is something like a Chosen select in which each option of the select has the order number, associated email address, and attendee names. Then you can type in any of those and it will come up. This would be especially nice if you can format the options to display in multiple rows (thinking of the Facebook search bar).

Alsoooo it would be really great if selecting them pulled up their whole order info, ajax style.

Or. It could be a multiselect and and you can select all the orders that you're marking as paid right now, all in one box!

**Comment by harrislapiroff** [comment]
Yeah, an order live filter would be badass. I guess I'm wondering if this is a common enough task, though, that there should be a form (for events that have check payment enabled) on the event dashboard for quick payment entry, basically.

**Comment by harrislapiroff** [comment]
That quick payment entry form could also have an ajax preview of the order though, to confirm that you had typed in a valid and correct order code. That'd be pretty sweet.

**Comment by nmorduch** [comment]
So Naomi had fun with box drawing and then they look sad here. Anyway, my thought is to have it in a little tabbed thing when checks are enabled because you want both *find* and *pay* options but you don't want to be tabbing through a lot of buttons; you just want to choose what mode you're in, then do a bunch of them.

```
┏━━━━━━┓───────────┐
┃ Find ┃ Mark Paid │
┃      ┗━━━━━━━━━━━┪
┃ ╔══════════╗ ┌─┐ ┃
┃ ║Find order║ │→│ ┃
┃ ╚══════════╝ └─┘ ┃
┗━━━━━━━━━━━━━━━━━━┛

┏━━━━━━┓───────────┐
┃ Find ┃ Mark Paid │
┃      ┗━━━━━━━━━━━┪
┃ ╔══════════╗ ┌─┐ ┃
┃ ║Ord       ║ │→│ ┃
┃ ╠══════════╣ └─┘ ┃
┃ ║Order 1   ║     ┃
┃ ╟──────────╢     ┃
┃ ║Order 2   ║     ┃
┃ ╟──────────╢     ┃
┃ ║Order 3   ║     ┃
┃ ╚══════════╝     ┃
┗━━━━━━━━━━━━━━━━━━┛
```

**Comment by nmorduch** [comment]
![messages image 1621604897](https://cloud.githubusercontent.com/assets/1051450/6045136/c2d3da00-ac65-11e4-8231-74382e96f749.png)

**Comment by harrislapiroff** [comment]
I dig it.

**Comment by nmorduch** [comment]
Sry I spent like years on the drawing but it was funnnn i used all the kinds of lines.

**Comment by harrislapiroff** [comment]
![250px-diglett_village_diglett](https://cloud.githubusercontent.com/assets/296631/6045172/145da2e8-ac66-11e4-9f29-3abaeb278908.png)

**Comment by nmorduch** [comment]
![messages image 1585257055](https://cloud.githubusercontent.com/assets/1051450/6045180/1d1851d0-ac66-11e4-8c9f-67a198e627f0.png)

**Comment by nmorduch** [comment]
Band name: All Kinds of Lines

**Comment by melinath** [comment]
I think having simple search-by-code is most widely applicable - i.e. we need to implement it anyway.

I really don't think it's intuitive to have a separate box (or a separate option on the same box) that lets you mark one or more orders as having their payment confirmed. I *do* think it's intuitive to have an ajax-y button on the table page that works just like on the detail page, or to have checkboxes that let you choose orders and mark them paid, or to have a "Mark all paid" button similar to "Mark all read" in GMail.

**Comment by harrislapiroff** [comment]
I think it is also important to have those things (search by code, actions, and a boolean checkbox of some sort) but I also think that entering checks might be a common enough task that it's worth having a "shortcut" for doing it.

I was trying to walk through the actions I would ideally take if I were an event treasurer and I have 20 checks in hand with order codes written on them for an event with, say, 200 attendees. I could browse the table and try to find those 20 records and check them all, but manually searching for records in a spreadsheet according to a random string _sucks_. I could search for each record using the table's search and then check the "paid" box and that would probably be fine.

But I was thinking that if I had my ideal interface for it, it would be one that was on the dashboard (maybe in the finance summary table) that was marked "record payment" and had a field for the order number. I could type in all or part of the order number, visually confirm that I had entered the right order, press enter, and then do the next check. I'd have all 20 done in no time.

**Comment by harrislapiroff** [comment]
Maybe I'm overestimating how common of a task this is. Maybe it will not be an oft-used feature and isn't worth the effort and space it would take on the summary page. That's totally possible. But I feel pretty sure I'm right about my imagined interface being faster and easier.

**Comment by melinath** [comment]
Okay, you've convinced me that it would make sense to have some kind of interface like this. But I'm concerned that it's gotten to be a pretty big feature.

What about having a separate tab under "orders" where you can just pull up a full list of orders with unconfirmed checks and do a cmd-f search of the page to find the order you're looking for, then click a button to confirm payment over ajax? This wouldn't scale to an event with 4,000 attendees, but it should work for events of ~200 people. And I think it would be easier to implement in the short term.

But maybe it would be better to just do it your way now. And maybe it wouldn't actually be that difficult. I don't know.

**Comment by melinath** [comment]
Just had someone else run into this.
